src/fairseq2/nn/checkpointing.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from functools import partial
+
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
```

Comment:
This is depending on an internal API. Do you know if PT plans to make the API public at some point ?

Reply:
PyTorch has some confusing API conventions. In this case the underscore prefix indicates that the API is more or less finalized, but the implementation is not stable yet. People are free to use such APIs on their own risk. In fact, just yesterday I realized that `torch.utils.checkpoint.checkpoint` heavily leaks memory when its `debug` parameter is set to `True`.

This particular `torch.distributed.algorithms._checkpoint` API has been out for several years now and used by pretty much all well-known training frameworks. I don't know why it is not marked as stable yet, most likely no one owns it at the moment and people already know and use it as it is.

Reply:
You are welcome!